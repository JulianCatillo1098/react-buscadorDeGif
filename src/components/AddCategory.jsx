import { useState } from "react";
import Proptypes from 'prop-types'

export const AddCategory = ({nuevoValor}) => {

  const [estadoInput, setEstadoInput] = useState("");

  const inputChange = ({target}) => {
    setEstadoInput(target.value);

  };
  
  const onSubmit = (eventoSubmit) => {
    eventoSubmit.preventDefault();
    if (estadoInput.trim().length <= 1) return;
    setEstadoInput('');
    
    nuevoValor(estadoInput.trim());
    
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder=" Buscar gif"
        value={estadoInput}
        onChange={(e)=>inputChange(e)}
      ></input>
    </form>
  );
};

AddCategory.prototype={
  nuevoValor: Proptypes.func.isRequired
}