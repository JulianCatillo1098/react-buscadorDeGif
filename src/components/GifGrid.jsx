import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import Proptypes from 'prop-types'

export const GifGrid = ({ category }) => {
  const{ imagenes,isLoading} = useFetchGifs (category)

  return (
    <>
    {isLoading && <h2>Cargando...</h2>}
      <h3>{category}</h3>
      <div className="card-grid">
        {imagenes.map((imagen) => {
          return <GifItem key={imagen.id} 
          // title={imagen.title} url={imagen.url}
          {...imagen}
          />
        })}
      </div>
    </>
  );
};

GifGrid.protoType={
  category : Proptypes.string.isRequired
}