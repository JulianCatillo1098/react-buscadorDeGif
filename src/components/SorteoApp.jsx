import { useState } from "react";



export const SorteoApp = () => {
    const [equipos, setEquipos] = useState(["pescara","merseyside red","juevntus","atletico","benfica","olympique de marcella","man blue","flamnego","tedloghec","north london","manchester united","barcelona","sc internacional","psv","atletico mineiro","bayer munich",]);
    const [equiposd, setEquiposd] = useState(["bosnia","colombia","brasil","suiza","chile","italia","serbia","bulgaria","costa rica ","francia","holanda","iran","inglaterra","argentina","polonia","venezuela",]);
      
  function generarPosicionesAleatorias(n) {
    const posiciones = Array.from(Array(n).keys());
    posiciones.sort(() => Math.random() - 0.5);
    return posiciones;
  }

  function generarPosicionesAleatoriasd(n) {
    const posicionesd = Array.from(Array(n).keys());
    posicionesd.sort(() => Math.random() - 0.5);
    return posicionesd;
  }

  function cambiarPosiciones() {
    const posiciones = generarPosicionesAleatorias(equipos.length);
    const nuevoArreglo = posiciones.map((posicion) => equipos[posicion]);
    setEquipos(nuevoArreglo);
  }
  function cambiarPosicionesd() {
    const posicionesd= generarPosicionesAleatoriasd(equiposd.length);
    const nuevoArreglod= posicionesd.map((posicion) => equiposd[posicion]);
    setEquiposd(nuevoArreglod);
  }

  return (
    <>
      <div>
        <button onClick={cambiarPosiciones}>Cambiar posiciones</button>
        <button onClick={cambiarPosicionesd}>Cambiar posiciones</button>
      </div>
      <div className="row d-flex">
        <h3>Grupo A</h3>
        <input value={equipos[0]} />
        <input value={equipos[1]} />
        <input value={equiposd[0]} />
        <input value={equiposd[1]} />
      </div>
      <div className="row d-flex">
        <h3>Grupo B</h3>
        <input value={equipos[2]} />
        <input value={equipos[3]} />
        <input value={equiposd[2]} />
        <input value={equiposd[3]} />
      </div>
      <div className="row d-flex">
        <h3>Grupo C</h3>
        <input value={equipos[4]} />
        <input value={equipos[5]} />
        <input value={equiposd[4]} />
        <input value={equiposd[5]} />
      </div>
      <div className="row d-flex">
        <h3>Grupo D</h3>
        <input value={equipos[6]} />
        <input value={equipos[7]} />
        <input value={equiposd[6]} />
        <input value={equiposd[7]} />
      </div>
      <div className="row d-flex">
        <h3>Grupo E</h3>
        <input value={equipos[8]} />
        <input value={equipos[9]} />
        <input value={equiposd[8]} />
        <input value={equiposd[9]} />
      </div>
      <div className="row d-flex">
        <h3>Grupo F</h3>
        <input value={equipos[10]} />
        <input value={equipos[11]} />
        <input value={equiposd[10]} />
        <input value={equiposd[11]} />
      </div>
      <div className="row d-flex">
        <h3>Grupo G</h3>
        <input value={equipos[12]} />
        <input value={equipos[13]} />
        <input value={equiposd[12]} />
        <input value={equiposd[13]} />
      </div>
      <div className="row d-flex">
        <h3>Grupo H</h3>
        <input value={equipos[14]} />
        <input value={equipos[15]} />
        <input value={equiposd[14]} />
        <input value={equiposd[15]} />
      </div>
    </>
  );
};
