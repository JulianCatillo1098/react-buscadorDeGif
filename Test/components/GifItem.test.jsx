import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('pruebas en GifItem', () => {

    const title='JULIAN'
    const url ='https://one-punch.com/julian.jpg' 

    test('debe de hacer match con el snapsho',() => {

        const { container } = render (<GifItem title={title} url = {url}/>)
        expect (container).toMatchSnapshot()
        
    
    })

    test('la img debe ter el src y el atl que estoy wnviando',()=>{
        render (<GifItem title={title} url={url}/>)
        // screen.debug() ----srive para ver que vamos a evaluar ---
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        // --- destructuramos las dos lineas anteriores ---
        const { src, alt} = screen.getByRole('img')
        expect (src).toBe(url)
        expect(alt).toBe(title)
        
    })

    test ('debe mostrar el tituko en el componente',()=>{
        render (<GifItem title={title} url={url}/>)
        expect (screen.getByText(title)).toBeTruthy()

    })

    test


})