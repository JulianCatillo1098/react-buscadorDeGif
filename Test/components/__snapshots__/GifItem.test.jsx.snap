// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifItem debe de hacer match con el snapsho 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="JULIAN"
      src="https://one-punch.com/julian.jpg"
    />
    <p>
      JULIAN
    </p>
  </div>
</div>
`;
